blocks/identity-block/features/account-management/_children/EmailEditableFieldContainer.jsx

```diff
@@ -1,30 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useFusionContext } from 'fusion:context';
 import getProperties from 'fusion:properties';
 import getTranslatedPhrases from 'fusion:intl';
+import { useIdentity } from '../../..';
 import EditableFieldPresentational from '../../../components/EditableFormInputField';
 import FormInputField, { FIELD_TYPES } from '../../../components/FormInputField';
 
-function EmailEditableFieldContainer({ email, onSubmit }) {
+function EmailEditableFieldContainer({ email }) {
+  const [error, setError] = useState(null);
+  const [emailAddress, setEmailAddress] = useState();
   const { arcSite } = useFusionContext();
   const { locale } = getProperties(arcSite);
   const phrases = getTranslatedPhrases(locale);
 
+  const { Identity } = useIdentity();
+
   const formEmailLabel = phrases.t('identity-block.email');
   const emailRequirements = phrases.t('identity-block.email-requirements');
   const editText = phrases.t('identity-block.edit');
+  const emailError = phrases.t('identity-block.update-email-error');
+
+  const handleEmailUpdate = ({ email: newEmail }) => Identity
+    .updateUserProfile({ email: newEmail })
+    .then((profileObject) => {
+      setEmailAddress(profileObject.email);
+      setError(null);
+    }).catch(() => {
+      setError(emailError);
+      throw new Error();
+    });
+
+  const handleCancelEdit = () => {
+    setError(null);
+  };
 
   return (
     <EditableFieldPresentational
-      initialValue={email}
+      initialValue={emailAddress || email}
```

Comment:
can we name these like incomingEmail or current email? email and emailAddress doesn't really help me understand it; likely just a merge issue